src/Platforms/AbstractPlatform.php

```diff
@@ -2416,7 +2416,7 @@ public function getCreateIndexSQL(Index $index, $table)
         $columns = $index->getColumns();
 
         if (count($columns) === 0) {
-            throw new InvalidArgumentException("Incomplete definition. 'columns' required.");
+            throw new InvalidArgumentException("Incomplete or invalid index definition `$name` on table `$table`");
```

Comment:
We don't seem to be using backticks for this, but rather, single quotes: https://github.com/doctrine/dbal/search?q=%22on+table%22
I can't say I like that a lot, IMO double quotes, but let's be consistent with what we have I suppose.

```suggestion
            throw new InvalidArgumentException("Incomplete or invalid index definition '$name' on table '$table'");
```

Also, could you please write a unit test for this?

Reply:
Updated the backticks to single quotes.

I'm not sure a unit test is appropriate here because:
1. I don't even know how you would test this or even write the test for this, especially considering I only stumbled upon this as a result of third party tools exposing it (symfony's maker/doctrine bridge), not to mention the doctrine/orm requirement.
2. https://softwareengineering.stackexchange.com/a/406096
3. Along with 2, if it was really that important to programmatically capture and handle the error in runtime, then an entirely new exception class would need to be created that encapsulates the table and index names, for what is essentially a "compile-time" error.

Reply:
1. Not sure what's blocking you: I'd create a new file called `tests/Platforms/AbstractPlatformTest.php`, then follow https://matthewsetter.com/getmockforabstractclass-overview/ . Another option would be to modify `tests/Platforms/AbstractPlatformTest.php` but I think it's a bad idea: we don't need to run that test on every platform, it's not a functional test.
2. I don't remember saying you should test the wording of this. If you do want to also test it, I'd recommend to just check that the name of the index and of the table are contained in the exception message.
3. Right. What we want to ensure is that an exception is thrown in that case, and one that can be understood by the end user, the type itself is not very important.

Reply:
FWIW, earlier we agreed to use double quotes in messages. See https://github.com/doctrine/dbal/pull/3233#discussion_r232740279 and https://github.com/doctrine/dbal/pull/3536 for reference.

Reply:
Now that I'm reviewing the rest of the file in more detail, rather than looking at only this specific method and keeping the existing formatting, there's a number of other exception messages that could be improved.

I'd be willing to take this on (perhaps in a different PR?), including the changes to the quotes/non-quotes. What do you think?

Reply:
It still would make sense to me if the quotes were removed from this patch before it gets merged. It needs to be retargeted against `3.5.x` anyways since it's not a bugfix.

I can rebase and update the patch for you if you want.